### PR TITLE
refactor: move confirmEmail request to xapi

### PR DESCRIPTION
**Pull request by ksavosteev**
Description:
https://virtocommerce.atlassian.net/browse/PT-12829

--

QA-test:

Demo-test:

Download artifact URL: https://vc3prerelease.blob.core.windows.net/packages/vc-theme-b2b-vue-1.36.0-pr-663-de96-de96ab7f.zip

### CODE REVIEW

File: client-app/pages/sign-up.vue [hunk 1]
```diff
@@ -249,8 +257,12 @@ const onSubmit = handleSubmit(async (data) => {
     });
   }
 
-  if (result.succeeded) {
+  if (result.succeeded && !result.requireEmailVerification) {
     router.push({ name: "Welcome" });
+  } else if (result.succeeded && result.requireEmailVerification) {
+    openPopup({
+      component: RegistrationConfirmEmailDialog,
```
Comment (NaMax66) [inline client-app/pages/sign-up.vue:264]: router.push({ name: "Welcome" })  after confirm

File: client-app/core/api/graphql/account/mutations/registerAccount/requestRegistration.graphql [hunk 2]
```diff
@@ -11,6 +11,7 @@ mutation requestRegistration($command: InputRequestRegistrationType!) {
     }
     result {
       succeeded
+      requireEmailVerification
```
Comment (asvishnyakov) [inline client-app/core/api/graphql/account/mutations/registerAccount/requestRegistration.graphql:14]: Remove unused field

File: client-app/core/api/graphql/account/mutations/confirmEmail/confirmEmail.graphql [hunk 3]
```diff
@@ -0,0 +1,10 @@
+mutation confirmEmail($command: InputConfirmEmailType!) {
+  confirmEmail(command: $command) {
+    succeeded
+    errors {
+      code
+      description
+      parameter
+    }
```
Comment (asvishnyakov) [inline client-app/core/api/graphql/account/mutations/confirmEmail/confirmEmail.graphql:8]: Remove unused fields

**Review by asvishnyakov — CHANGES_REQUESTED** [review]

File: client-app/core/api/graphql/account/mutations/registerAccount/requestRegistration.graphql (hunk 2 above)
Comment (ksavosteev) [reply client-app/core/api/graphql/account/mutations/registerAccount/requestRegistration.graphql:14]: May be used in the future

File: client-app/core/api/graphql/account/mutations/confirmEmail/confirmEmail.graphql (hunk 3 above)
Comment (ksavosteev) [reply client-app/core/api/graphql/account/mutations/confirmEmail/confirmEmail.graphql:8]: May be used in the future

**Review by Andrew-Orlov — APPROVED** [review]

**Review by asvishnyakov — APPROVED** [review]

**Pull request merged**